tracer/src/Datadog.Trace/Configuration/ImmutableAzureAppServiceSettings.cs

```diff
@@ -191,16 +191,15 @@ private static bool ShouldSkipClientSpanWithinFunctions(Scope? scope)
             return null;
         }
 
-        public static bool GetIsFunctionsAppConsumptionPlan(IConfigurationSource source, IConfigurationTelemetry telemetry)
+        public static bool GetIsFunctionsAppUsingMiniAgent(IConfigurationSource source, IConfigurationTelemetry telemetry)
         {
             var config = new ConfigurationBuilder(source, telemetry);
-            var websiteSKU = config.WithKeys(ConfigurationKeys.AzureAppService.WebsiteSKU).AsString();
 
             var functionsExtensionVersion = config.WithKeys(ConfigurationKeys.AzureAppService.FunctionsExtensionVersionKey).AsString();
             var functionsWorkerRuntime = config.WithKeys(ConfigurationKeys.AzureAppService.FunctionsWorkerRuntimeKey).AsString();
             var isFunctionApp = functionsExtensionVersion is not null || functionsWorkerRuntime is not null;
 
-            return websiteSKU is "Dynamic" or null && isFunctionApp;
+            return isFunctionApp && (Environment.OSVersion.Platform == PlatformID.Unix || config.WithKeys(ConfigurationKeys.AzureAppService.WebsiteSKU).AsString() is "Dynamic" or null);
```

Comment:
For code readability, consider keeping the `websiteSKU` local variable instead of inlining it into the `return` statement.

```suggestion
            var websiteSKU = config.WithKeys(ConfigurationKeys.AzureAppService.WebsiteSKU).AsString();
            return isFunctionApp && (Environment.OSVersion.Platform == PlatformID.Unix || websiteSKU  is "Dynamic" or null);
```

Reply:
And maybe we can add a short comment explaining the logic here? Something about the fact that we always use the mini-agent for Function apps on Linux, but not always on Windows (and why). Thanks!